Adjustment to UploadReview logic to allow a super agency to publish for their child agencies

mxosman opened the pull request:
## Description of the change

Now that the BE sends reports' metadata, this updates the `UploadReview` logic to utilize the metadata for rendering instead of the report IDs + querying the `reportStore`.

## Related issues

Closes #653 

## Checklists

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

lilidworkin reviewed (commented):
Nice! Clean and simple change. Two things: 
1) I think we should wait to merge this and do it at the same time as the BE PR gets in, right, since it's a breaking change?
2) The BE PR also added "agency_name" to both the Datapoint and Report objects. I know the FE isn't using that info yet (I think I may have Ivan work on that) but can you add that field as optional to both types, and verify via console.log or something that you're able to access it? TY! Will accept after that.

mxosman commented:
>Two things:
> 
> 1. I think we should wait to merge this and do it at the same time as the BE PR gets in, right, since it's a breaking change?

Absolutely!

> 2. The BE PR also added "agency_name" to both the Datapoint and Report objects. I know the FE isn't using that info yet (I think I may have Ivan work on that) but can you add that field as optional to both types, and verify via console.log or something that you're able to access it? TY! Will accept after that.

Updated both datapoint and report overview types to include `agency_name`. Can confirm that both are accessible!

<img width="826" alt="Screenshot 2023-05-24 at 12 04 51 PM" src="https://github.com/Recidiviz/justice-counts/assets/59492998/fe926c1b-e343-422c-a13c-2f8fb9713f9e">
<img width="791" alt="Screenshot 2023-05-24 at 12 03 42 PM" src="https://github.com/Recidiviz/justice-counts/assets/59492998/343fe96d-adc8-4f03-83d5-5ebfc3971c86">

lilidworkin reviewed (approved):
Perfect, thank you!
